### PR TITLE
improve(serverless-orchestration): migrate to hardhat

### PR DESCRIPTION
**Pull request by mrice32**
**Motivation**

All packages should be fully migrated to running on hardhat without the truffle plugin.

**Summary**

These packages still require an external node to be run (either hardhat node or ganache), but they now don't require truffle.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [x]  All existing tests pass
- [ ]  Untested

**Issue(s)**

N/A

### CODE REVIEW

File: packages/serverless-orchestration/test/ServerlessHub.js [hunk 1]
```diff
@@ -69,15 +72,24 @@ contract("ServerlessHub.js", function (accounts) {
...
+    const node = ganache.server({ _chainIdRpc: chainId });
+    node.listen(port);
+    return new Web3("http://127.0.0.1:" + port);
+  };
+
   before(async function () {
+    accounts = await web3.eth.getAccounts();
+    [contractDeployer] = accounts;
     defaultChainId = await web3.eth.getChainId();
+    await runDefaultFixture(hre);
```
Comment (chrismaree) [inline packages/serverless-orchestration/test/ServerlessHub.js:85]: ideally, in the future, we should move away from these kinds of fixtures as they ended up slowing things down more than need be by deploying more than required a lot of the time.

**Review by chrismaree — APPROVED** [review]
This is great! awesome work on converting this package. I'm sure it's _much much_ faster now. I suspected there to be more pain with converting things with the network URLs but looks like hardhat is now pretty similar with the `web3.currentProvider.host to `network.config.url`.

**Review by daywiss — APPROVED** [review]
lgtm

File: packages/serverless-orchestration/test/ServerlessHub.js (hunk 1 above)
Comment (mrice32) [reply packages/serverless-orchestration/test/ServerlessHub.js:85]: Yep! There are a bunch of simplicity-vs-speed tradeoffs here. I basically just use this when any tests (intentionally or not) depend on pre-deployed contracts. I don't include it by default, though, so most of our tests actually don't need it, which means we get most of this speedup already!

**Pull request merged**